Feat(eos_designs): Add platform settings for 7368X4

gmuloc opened the pull request:
## Change Summary

Add support for platform specific default variable (management interface and reload delay counters for mlag and non-mlag) for 7368X4 series.

## Related Issue(s)

Fixes #1680

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Added an entry for platform series 7368X4 in the `roles/eos_designs/defaults/main/main.yml` file reflecting the platform defaulrs:

```
  - platforms: [ '7368X4']
    management_interface: Management0
    reload_delay:
      mlag: 300
      non_mlag: 330
```

* 7368X4 is a modular chassis and its management interface is Management0
* The HW platform timers for 7368X4 are mlag 300ms and non-mlag 300ms. In order to follow the EVPN guide best practices, the non-mlag timer is set by default for this platform to 330ms.
* Test added in molecule to generate the config to verify both settings
* Documentation updated

## How to test

Molecule test added to generate two configurations for platform 7368X4 type to verify both management interface and reload delay counters

## Checklist

### User Checklist

<!-- Add your own checklist using MD syntax and by replacing N/A -->
- N/A

### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

gmuloche committed:
Feat(eos_designs): Add 7368X4 support

* 7368X4 is a modular chassis and its management interface is
  Management0
* The HW platform timers for 7368X4 are mlag 300ms and non-mlag 300ms.
  In order to follow the EVPN guide best practices, the non-mlag timer
  is set by default for this platform to 330ms.
* Test added in molecule to generate the config to verify both settings
* Documentation updated

tgodaA commented on ansible_collections/arista/avd/roles/eos_designs/defaults/main/main.yml:
```diff
@@ -111,6 +111,11 @@ platform_settings:
     reload_delay:
       mlag: 900
       non_mlag: 1020
+  - platforms: [ '7368X4']
```
Space needed at the end for consistency. **[ '7368X4' ]**

gmuloc replied:
interesting - it passed the pre-commit check like this. Fixed in next commit

tgodaA commented on ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/TESTS_PLATFORM.yml:
```diff
@@ -0,0 +1,20 @@
+      mlag_interfaces: [ Ethernet3 ]
+      mlag_peer_vlan: 4091
+      mlag_peer_ipv4_pool: 192.168.255.0/24
+      rack:
```
Any reason to keep this blank. Please remove this if not required.

gmuloc replied:
Removed it.

gmuloc committed:
Modifications after PR review

* Fix yaml syntax to match repo convention
* Remove blank group variable
* Fix management ip addresses in group variables to match the commited
  molecule result (missed in previous commit)

ClausHolbechArista commented on ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/hosts.yml:
```diff
@@ -85,6 +86,12 @@ all:
                     mgmt_interface_fabric:
                     mgmt_interface_host:
                     mgmt_interface_platform:
+            TESTS_PLATFORM:
```
do we need all this to test a new platform? Normally we just assign the platform to one of the existing devices in any scenario

gmuloc replied:
it is needed to check for MLAG counters as I need a group of 2 hosts to generate the mlag config - I tried to include it in the L2Leaf under TESTS but could not succeed - happy to change if there is a way. The advantage of cretaing this other node group is that it makes it possible to get some group_vars and so to generate smaller configs.

Of course happy to change it if there is a better way

gmuloc replied:
Following our discussions just pushed a new commit leveraging DC1-CL1A to test for the new platform variables hence reducing the new amount of generated configruation

gmuloc committed:
Leverage existing DC1-CL1A for molecule tests

* Rather than creating two new devices to test the platform addition,
  leverage the existin DC1-CL1A leaf (DC1-CL1B leaf still validates the
  default platform settings)

ClausHolbechArista reviewed (approved):
LGTM

tgodaA reviewed (approved):
LGTM
